src/main/java/org/jabref/gui/mergeentries/newmergedialog/fieldsmerger/GroupMerger.java

```diff
@@ -1,31 +1,38 @@
 package org.jabref.gui.mergeentries.newmergedialog.fieldsmerger;
 
-import java.util.Arrays;
-import java.util.regex.Pattern;
-import java.util.stream.Collectors;
+import java.util.Objects;
 
+import org.jabref.model.entry.BibEntryPreferences;
+import org.jabref.model.entry.KeywordList;
 import org.jabref.model.entry.field.StandardField;
 import org.jabref.model.strings.StringUtil;
 
 /**
  * A merger for the {@link StandardField#GROUPS} field
  * */
 public class GroupMerger implements FieldMerger {
-    public static final String GROUPS_SEPARATOR = ", ";
-    public static final Pattern GROUPS_SEPARATOR_REGEX = Pattern.compile("\s*,\s*");
+//    public static final String GROUPS_SEPARATOR = ", ";
+//    public static final Pattern GROUPS_SEPARATOR_REGEX = Pattern.compile("\s*,\s*");
+
+    private final BibEntryPreferences bibEntryPreferences;
+
+    public GroupMerger(BibEntryPreferences bibEntryPreferences) {
+        Objects.requireNonNull(bibEntryPreferences);
+        this.bibEntryPreferences = bibEntryPreferences;
```

Comment:
```suggestion
        this.bibEntryPreferences = Objects.requireNonNull(bibEntryPreferences);
```

I believe '@Nonnull` annotation does the same. Not sure though.